ichnaea/data/tests/test_station.py

```diff
@@ -240,45 +245,47 @@ def test_blocklist_skip(self, celery, redis, session):
+                obs2 = self.obs_factory(
+                    lat=obs.lat - 0.0003, source=source, **self.key(obs)
+                )
+                obs3 = self.obs_factory(
+                    lon=obs.lon + 0.0002, source=source, **self.key(obs)
+                )
+                obs4 = self.obs_factory(
+                    lon=obs.lon - 0.0004, source=source, **self.key(obs)
+                )
+                self.queue_and_update(celery, [obs, obs1, obs2, obs3, obs4])
+
+                self.check_areas(celery, [obs])
+                station = self.get_station(session, obs)
+                assert round(station.lat, 7) == round(obs.lat - 0.00004, 7)
+                assert round(station.max_lat, 7) == round(obs.lat + 0.0001, 7)
+                assert round(station.min_lat, 7) == round(obs.lat - 0.0003, 7)
+                assert round(station.lon, 7) == round(obs.lon - 0.00004, 7)
+                assert round(station.max_lon, 7) == round(obs.lon + 0.0002, 7)
+                assert round(station.min_lon, 7) == round(obs.lon - 0.0004, 7)
+                assert station.radius == 38
+                assert station.region == "GB"
+                assert station.samples == 5
+                assert station.source == source
+                assert round(station.weight, 2) == 5.0
+                self.check_blocked(station, None)
+                self.check_dates(station, self.today, self.today, self.today)
+
+            # Assert generated metrics are correct
+            assert mm.has_record(
+                "incr", "data.observation.insert", value=5, tags=[self.type_tag]
             )
-            obs4 = self.obs_factory(
-                lon=obs.lon - 0.0004, source=source, **self.key(obs)
+            assert mm.has_record(
+                "incr", "data.station.new", value=1, tags=[self.type_tag]
             )
-            self.queue_and_update(celery, [obs, obs1, obs2, obs3, obs4])
-
-            self.check_areas(celery, [obs])
-            station = self.get_station(session, obs)
-            assert round(station.lat, 7) == round(obs.lat - 0.00004, 7)
-            assert round(station.max_lat, 7) == round(obs.lat + 0.0001, 7)
-            assert round(station.min_lat, 7) == round(obs.lat - 0.0003, 7)
-            assert round(station.lon, 7) == round(obs.lon - 0.00004, 7)
-            assert round(station.max_lon, 7) == round(obs.lon + 0.0002, 7)
-            assert round(station.min_lon, 7) == round(obs.lon - 0.0004, 7)
-            assert station.radius == 38
-            assert station.region == "GB"
-            assert station.samples == 5
-            assert station.source == source
-            assert round(station.weight, 2) == 5.0
-            self.check_blocked(station, None)
-            self.check_dates(station, self.today, self.today, self.today)
-
-        stats.check(
-            counter=[
-                ("data.observation.insert", 2, [self.type_tag]),
-                ("data.station.new", 2, [self.type_tag]),
```

Comment:
These assert that there are two of each of these messages. I need to reimplement that.